src/bin/pg_autoctl/pgctl.c

```diff
@@ -918,6 +1030,100 @@ escape_recovery_conf_string(char *destination, const char *recoveryConfString)
 }
 
 
+/*
+ * prepare_primary_conninfo
+ */
+static int
+prepare_primary_conninfo(char *primaryConnInfo, int primaryConnInfoSize,
+						 const char *primaryHost, int primaryPort,
+						 const char *replicationUsername,
+						 const char *replicationPassword)
+{
+	char primaryConnInfoRaw[MAXCONNINFO] = { 0 };
+	char *connInfoEnd = primaryConnInfoRaw;
+
+ 	connInfoEnd += make_conninfo_field_str(connInfoEnd, "host", primaryHost);
+	connInfoEnd += make_conninfo_field_int(connInfoEnd, "port", primaryPort);
+	connInfoEnd +=
+		make_conninfo_field_str(connInfoEnd, "user", replicationUsername);
+
+	if (replicationPassword != NULL)
+	{
+		connInfoEnd +=
+			make_conninfo_field_str(connInfoEnd, "password", replicationPassword);
+	}
+
+	return escape_recovery_conf_string(primaryConnInfoRaw, primaryConnInfo);
```

Comment:
we keep appending to `primaryConnInfoRaw` by using connInfoEnd. At this line we completely override it value coming from `primaryConnInfo`. Could you check ?

Reply:
you probably only need to reverse parameters here